adapters/dmx/dmx.go

```diff
@@ -77,6 +87,12 @@ func (adapter *DmxAdapter) MakeRequests(request *openrtb.BidRequest, req *adapte
 		appPublisherCopy := *request.App.Publisher
 		dmxReq.App = &appCopy
 		dmxReq.App.Publisher = &appPublisherCopy
+		if dmxReq.App.Publisher.ID == "" {
+			dmxReq.App.Publisher.ID = publisherId
+		}
+		dmxRawPubId.Dmx.Id = UserSellerOrPubId(rootExtInfo.Bidder.PublisherId, rootExtInfo.Bidder.MemberId)
+		ext, _ := json.Marshal(dmxRawPubId)
```

Comment:
Thats not exactly what we had in mind. :)

If there is a Marshal error, would you be ok sending an empty ext to your bidding server or would you consider this an invalid request?

Although unlikely to occur, just in case it does the code paths should be supportive in bubbling up marshal errors.

Reply:
Yes this value can be empty toosince we have several check

Reply:
Gotcha. I've been giving it more thought. Since a Marshal failure is an indication of something going very wrong in the server, I think it makes to bubble up the error. That would look something like this:
```
ext, err := json.Marshal(dmxRawPubId)
if err != nil {
  errs = append(errs, fmt.Errorf("unable to marshal ext, %v", err))
  return nil, errs. <-- or maybe just do this at the end.
}
```

Thoughts @hhhjort @guscarreon?

Reply:
I think it would be a good idea to bubble up the error. Unlikely to happen, but will greatly ease debugging if and when it does happen.